google/cloud/storage/examples/storage_async_samples.cc

```diff
@@ -285,9 +294,61 @@ void OpenObjectReadFullObject(
   std::cout << "The range contains " << count << " newlines\n";
 }
 
+void OpenMultipleObjectRangedRead(
+    google::cloud::storage_experimental::AsyncClient& client,
+    std::vector<std::string> const& argv) {
+  //! [open-multiple-object-ranged-read]
+  // [START storage_open_multiple_object_ranged_read]
+  namespace gcs_ex = google::cloud::storage_experimental;
+
+  // Helper coroutine to count newlines returned by an AsyncReader.
+  auto count_newlines =
+      [](gcs_ex::AsyncReader reader,
+         gcs_ex::AsyncToken token) -> google::cloud::future<std::uint64_t> {
+    std::uint64_t count = 0;
+    while (token.valid()) {
+      auto [payload, t] = (co_await reader.Read(std::move(token))).value();
+      token = std::move(t);
+      for (auto const& buffer : payload.contents()) {
+        count += std::count(buffer.begin(), buffer.end(), '\n');
+      }
+    }
+    co_return count;
+  };
+  auto coro = [&count_newlines](
+                  gcs_ex::AsyncClient& client, std::string bucket_name,
+                  std::string object_name1, std::string object_name2,
+                  std::string object_name3) -> google::cloud::future<void> {
+    // List of object names to read (passed as arguments)
+    std::vector<std::string> object_names = {object_name1, object_name2,
+                                             object_name3};
+    std::vector<google::cloud::future<std::uint64_t>> futures;
+
+    // Start ranged reads for all objects and collect futures
+    // This example opens multiple objects, not one object multiple times.
+    for (auto const& name : object_names) {
+      auto descriptor =
+          (co_await client.Open(gcs_ex::BucketName(bucket_name), name)).value();
+      auto [reader, token] = descriptor.Read(0, 1024);
+      futures.push_back(count_newlines(std::move(reader), std::move(token)));
+    }
+
+    // Wait for all futures and print results
```

Comment:
nit: I think we should not be waiting for all the futures to be ready, we should just start reading the futures which are ready.